File: 1305_All_Elements_In_Two_Binary_Search_Trees.java
/*
Runtime: 16 ms, faster than 67.63% of Java online submissions for All Elements in Two Binary Search Trees.
Memory Usage: 42.3 MB, less than 83.75% of Java online submissions for All Elements in Two Binary Search Trees.
*/ 

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
import java.util.*; 

class Solution {
    
    public static void traverseTree(TreeNode root, List<Integer> listOfNodes)
    {
        if(root == null) return;  
        listOfNodes.add(root.val); 
        traverseTree(root.left, listOfNodes); 
        traverseTree(root.right, listOfNodes); 
        return; 
    }
    
    public List<Integer> getAllElements(TreeNode root1, TreeNode root2) {
        List<Integer> listOfNodes = new ArrayList<Integer>(); 
        traverseTree(root1, listOfNodes);
        traverseTree(root2, listOfNodes); 
        Collections.sort(listOfNodes);
        return listOfNodes;  
    }
}

File: 1305_All_Elements_In_Two_Binary_Search_Trees.py
'''
Runtime: 344 ms, faster than 92.99% of Python3 online submissions for All Elements in Two Binary Search Trees.
Memory Usage: 20.2 MB, less than 67.99% of Python3 online submissions for All Elements in Two Binary Search Trees.
'''

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    
    def traverseTree(self, root, listOfNodes):
        if(root == None):
            return listOfNodes 
        listOfNodes.append(root.val)
        if (root.left != None):
            self.traverseTree(root.left, listOfNodes)
        if (root.right != None):
            self.traverseTree(root.right, listOfNodes)
        return listOfNodes 
        
    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> List[int]:
        rootOne = self.traverseTree(root1, [])
        rootTwo = self.traverseTree(root2, []) 
        rootOne.extend(rootTwo)
        return sorted(rootOne)
        
